frontend/packages/kubevirt-plugin/src/components/vm-snapshots/snapshot-details.tsx

```diff
@@ -0,0 +1,171 @@
+import { DetailsPage } from '@console/internal/components/factory';
+import {
+  useK8sWatchResource,
+  WatchK8sResource,
+} from '@console/internal/components/utils/k8s-watch-hook';
+import snapshotRestoreModal from '../modals/snapshot-restore-modal/snapshot-restore-modal';
+import { VirtualMachineRestoreModel, VirtualMachineSnapshotModel } from '../../models';
+import { VMRestore, VMSnapshot } from '../../types';
+import {
+  getVmRestoreSnapshotName,
+  getVmRestoreTime,
+  getVmSnapshotVmName,
+  isVMSnapshotReady,
+} from '../../selectors/snapshot/snapshot';
+import { VMSnapshotStatus } from './snapshot-row';
+
+const { editYaml } = navFactory;
+const { common } = Kebab.factory;
+const menuActions = [
+  (kind, snapshot) => ({
+    label: 'Restore Snapshot',
+    isDisabled: !isVMSnapshotReady(snapshot),
+    callback: () => snapshotRestoreModal({ snapshot }),
+  }),
+  ...common,
+];
+
+const SnapshotDetails: React.FC<DetailsProps> = ({ obj, customData: { restores } }) => {
+  const relevantRestore = restores[getName(obj)];
+  return (
+    <>
+      <div className="co-m-pane__body">
+        <SectionHeading text="Snapshot Details" />
+        <div className="row">
+          <div className="col-sm-6">
+            <ResourceSummary resource={obj} showAnnotations={false}>
+              {obj?.metadata?.annotations?.description && (
+                <>
+                  <dt>Description</dt>
+                  <dd>{obj.metadata.annotations.description}</dd>
+                </>
+              )}
+            </ResourceSummary>
+          </div>
+          <div className="col-sm-6">
+            <dl className="co-m-pane__details">
+              <DetailsItem label="Status" obj={obj} path="status">
+                <VMSnapshotStatus snapshot={obj} restore={relevantRestore} />
+              </DetailsItem>
+              <dt>Last Restored</dt>
+              <dd>
+                {relevantRestore ? (
+                  <Timestamp timestamp={getVmRestoreTime(relevantRestore)} />
+                ) : (
+                  DASH
+                )}
+              </dd>
+            </dl>
+          </div>
+        </div>
+      </div>
+      <div className="co-m-pane__body">
+        <SectionHeading text="Conditions" />
+        <Conditions conditions={obj.status?.conditions} />
+      </div>
+    </>
+  );
+};
+
+export const SnapshotDetailsPage: React.FC<SnapshotDetailsPageProps> = ({ match, ...props }) => {
+  const { ns: namespace, name: snapshotName } = match.params;
+
+  const resource: FirehoseResource = {
+    kind: VirtualMachineSnapshotModel.kind,
+    prop: 'snapshot',
+    isList: false,
+    name: snapshotName,
+    namespace,
+  };
+
+  const restoreResource: WatchK8sResource = {
+    isList: true,
+    kind: VirtualMachineRestoreModel.kind,
+    namespaced: true,
+    namespace,
+  };
+
+  const [restores, restoresLoaded] = useK8sWatchResource<VMRestore[]>(restoreResource);
+
+  const mappedRelevantRestores = React.useMemo(
```

Comment:
could we make a hook for this, since we are using ti twice?

Reply:
you could include `useK8sWatchResource` into the new hook and it would return `[mappedRestores, loaded, error]`